brain/python/neuron/morphology.cpp

```diff
@@ -0,0 +1,227 @@
+}
+
+list Soma_getCentroid( const SomaWrapper& soma )
+{
+    list result;
+    const Vector3f& centroid = soma.getCentroid();
+    result.append( centroid[0] );
+    result.append( centroid[1] );
+    result.append( centroid[2] );
+    return result;
+}
+
+#define GET_SECTION_ARRAY( Array ) \
+object Section_get##Array( const SectionWrapper& section ) \
+{                                                   \
+    return toNumpy( section.get##Array( ));         \
+}
+
+GET_SECTION_ARRAY( Samples )
+GET_SECTION_ARRAY( SampleDistancesToSoma )
+
+object Section_getSamplesFromPositions( const SectionWrapper& section,
+                                        object points )
+{
+    const floats pointVector = vectorFromPython< float >(
+        points, "Cannot convert argument to float vector");
+    return toNumpy( section.getSamples( pointVector ));
+}
+
+object Section_getParent( const SectionWrapper& section )
+{
+    if( section.hasParent( ))
+        return object( SectionWrapper( section.getParent( ),
+                                       section.morphology ));
+    else
+        return object();
+}
+
+object Section_getChildren( const SectionWrapper& section )
+{
+    const Sections& sections = section.getChildren();
+    list result;
+    for ( const auto& i : sections )
+        result.append( SectionWrapper( i, section.morphology ));
+    return result;
+}
+
+MorphologyPtr createMorphology( const std::string& uri )
+{
+    return MorphologyPtr( new Morphology( URI( uri )));
+}
+
+MorphologyPtr createMorphologyWithTransform( const std::string& uri,
+                                             object transform )
+{
+    return MorphologyPtr( new Morphology( URI( uri ), fromNumpy( transform )));
+}
+
+#define GET_MORPHOLOGY_ARRAY( Array ) \
+object Morphology_get##Array( const MorphologyPtr& morphology ) \
+{                                                               \
+    return toNumpy( morphology->get##Array(), morphology );     \
+}
+
+GET_MORPHOLOGY_ARRAY( Points )
+GET_MORPHOLOGY_ARRAY( Sections )
+GET_MORPHOLOGY_ARRAY( Apicals )
+
+object Morphology_getSectionIDs( const MorphologyPtr& morphology, object types )
+{
+    const SectionTypes typeVector = vectorFromPython< SectionType >(
+        types, "Cannot convert argument to SectionType list");
+    return toNumpy( morphology->getSectionIDs( typeVector ));
+}
+
+object Morphology_getSectionTypes( const MorphologyPtr& morphology )
+{
+    return toNumpy( morphology->getSectionTypes( ), morphology );
+}
+
+object Morphology_getSectionsByType( const MorphologyPtr& morphology,
+                                     object types )
+{
+    SectionTypes typeVector;
+    try
+    {
+        extract< SectionType > extractor( types );
+        if ( extractor.check( ))
+        {
+            typeVector.push_back(( SectionType )extractor );
+        }
+        else
+        {
+            stl_input_iterator< SectionType > i(types), end;
+            for ( ; i != end; ++i)
```

Comment:
Probably I'll miss some. This style is so against common use typography... and no, I'm not convinced that it's superior to the normal style.